tools/e2etesting/DeploySimulation.ps1

```diff
@@ -1,116 +1,160 @@
     "fastNodeRate" = 1
     "fastNodeType" = "uint"
-    "iotEdgeConnString" = $edgeDeviceConnectionString.connectionString
-    "keyVaultName" = $keyVaultName
 }
 
-Write-Host "Preparing to deploy e2e.edge.and.plc.simulation.json"
+Write-Host "Preparing to deploy e2e.plc.simulation.json"
+
+$plcTemplate = [System.IO.Path]::Combine($TemplateDir, "e2e.plc.simulation.json")
+$plcDeployment = New-AzResourceGroupDeployment -ResourceGroupName $ResourceGroupName -TemplateFile $plcTemplate -TemplateParameterObject $plcTemplateParameters
 
-# Execute deployment using ARM template
-$template = [System.IO.Path]::Combine($templateDir, "e2e.edge.and.plc.simulation.json")
-$simulationDeployment = New-AzResourceGroupDeployment -ResourceGroupName $resourceGroupName -TemplateFile $template -TemplateParameterObject $templateParameters
-if ($simulationDeployment.ProvisioningState -ne "Succeeded") {
-    Write-Error "Deployment $($simulationDeployment.ProvisioningState)." -ErrorAction Stop
+if ($plcDeployment.ProvisioningState -ne "Succeeded") {
+    Write-Error "Deployment $($plcDeployment.ProvisioningState)." -ErrorAction Stop
 }
 
-# Generate URLs to access published_nodes.json via HTTP endpoint of plc-simulation.
+Write-Host "Getting IPs of ACIs for simulated PLCs..."
+
+$containerInstances = Get-AzContainerGroup -ResourceGroupName $ResourceGroupName
+
 $plcSimNames = ""
-for ($i=1; $i -le $numberOfSimulations; $i++) {
-    $aci = az container show --resource-group $resourceGroupName --name "$($simulationDeployment.Outputs.plcPrefix.Value)$($i)" | ConvertFrom-Json
-    $plcSimNames += "$($aci.ipAddress.fqdn);"
+foreach ($ci in $containerInstances) {
+    $plcSimNames += $ci.Fqdn + ";"
 }
 
-# Store Url to PLC simulations in KeyVault
-$secretvalue = ConvertTo-SecureString $plcSimNames -AsPlainText -Force
-$secret = Set-AzKeyVaultSecret -VaultName $keyVaultName -Name 'plc-simulation-urls' -SecretValue $secretvalue
-if (-Not $secret.Enabled) {
-    Write-Error "Couldn't store URLs to access PLC simulation in KeyVault" -ErrorAction Stop
+Write-Host "Adding/Updating KeVault-Secret 'plc-simulation-urls' with value '$($plcSimNames)'..."
+Set-AzKeyVaultSecret -VaultName $keyVault.VaultName -Name 'plc-simulation-urls' -SecretValue (ConvertTo-SecureString $plcSimNames -AsPlainText -Force) | Out-Null
+
+Write-Host "Getting Device Connection String for IoT Edge Deployment..."
+$edgeDeviceConnectionString = Get-AzIotHubDeviceConnectionString -ResourceGroupName $ResourceGroupName -IotHubName $iotHub.Name -DeviceId $edgeIdentity.Id -KeyType primary
+$edgeDeviceConnectionString = $edgeDeviceConnectionString.ConnectionString
+
+$dnsPrefix = "e2etesting-edgevm-" + $testSuffix
+
+Write-Host "Using DNS prefix: $($dnsPrefix)"
+
+$edgeParameters = @{
+    "dnsLabelPrefix" = $dnsPrefix
+    "adminUsername" = $edgeVmUsername
+    "deviceConnectionString" = $edgeDeviceConnectionString
+    "authenticationType" = "password"
+    "adminPasswordOrKey" = $edgeVmPassword
```

Comment:
setting vmSize to "Standard_D2s_v3" is missing

Reply:
Reintroduced parameter.